system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -1593,8 +1644,58 @@ class CLdapClient : implements ILdapClient, public CInterface
 
             createLdapBasedn(NULL, m_ldapconfig->getUserBasedn(), PT_ADMINISTRATORS_ONLY);
             createLdapBasedn(NULL, m_ldapconfig->getGroupBasedn(), PT_ADMINISTRATORS_ONLY);
+
+            //Create the HPCC Administrators group and admin user
+            if (!adminGroupName.isEmpty())
+            {
+                //Create HPCC admin group
+                bool groupExisted = organizationalUnitExists(m_ldapconfig->getAdminGroupDN());
+                if (!groupExisted)
+                {
+                    DBGLOG("Adding HPCC Admin group %s", adminGroupName.str());
+                    try { addGroup(adminGroupName.str(), nullptr, "HPCC Administrators"); }
+                    catch(...) {}//group may already exist, so just move on
+
+                    //Create HPCC Admin user
+                    const char * pUser = m_ldapconfig->getHPCCAdminUser_username();
+                    if (pUser)
+                    {
+                        DBGLOG("Creating HPCC Admin user %s", pUser);
+                        Owned<ISecUser> user = new CLdapSecUser(pUser, nullptr);
+                        user->credentials().setPassword(m_ldapconfig->getHPCCAdminUser_password());
+                        try { addUser(*user.get()); }
+                        catch(...) {}//user may already exist, so just move on
+
+                        //Add HPCC Admin user to admin group
+                        DBGLOG("Adding HPCC Admin user %s to HPCC Admin group %s", pUser, adminGroupName.str());
+                        try { changeUserGroup("add", pUser, adminGroupName); }
+                        catch(...) {}//user may already be in group so just move on
+                    }
+                    else
+                        DBGLOG("HPCC Admin user not specified in configuration (hpccAdminSecretKey)");
+                }
+            }
             createdOU = true;
         }
+
+        //Set ECLWatch permissions for HPCC admin group
+        if (!adminGroupName.isEmpty())
+        {
+            //SmcAccess OU attribute is not created until later in
```

Comment:
Seems like common code may be assuming we're in EclWatch here?  Is there a way this can be driven by the Authmap config or something else so no assumptions are made.

If I'm running other services and not EclWatch is this going to happen?

Reply:
Currently, this code to create default OUs has always been called by any LDAP client that includes the OU definitions in the config. So, this has always been done by ECLWatch, Dali, ECLCCServer, ECLRepository and various ESP Services.  To fix his would require fixing the OU creation as well. The proper fix is for each client that uses LDAP resource to ensure the LDAP resources they require are created, instead of trying to brute force all of them here.
That being said, this code is safe in that it won't break anything, it's just an unnecessary LDAP call on startup.  I recommend leaving it as is for now

Reply:
Ok, since the primary functionality already exists.